Bottom: 16ce99869f4fdc5a471cfe4f5c61f5c63b26c71c
Top:    d50babdc10223754a9011b7060b263fceb25230f
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-11-11 05:26:04 +0000

We can just delete kicked out holes

Since we don't keep holes from broken contours, if a contour is
kicked out, we know it is destined for the bin.


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 043db86..6aa18b2 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1351,6 +1351,7 @@ find_inside (const BoxType *b, void *cl)
   if (check->Flags.orient == PLF_DIR) {
     return 0;
   }
+
   if (poly_ContourInContour (info->want_inside, check)) {
     info->result = check;
     longjmp (info->jb, 1);
@@ -1459,7 +1460,8 @@ InsertHoles (jmp_buf * e, POLYAREA * dest, PLINE ** src)
 	}
       else
 	{
-          /* Need to check if this new hole means we need to kick out any old ones for reprocessing */
+          /* Need to check if this new hole means we need to kick out any old
+           * ones for deletion */
           while (1) {
             struct find_inside_info info;
             PLINE *prev;
@@ -1487,10 +1489,7 @@ InsertHoles (jmp_buf * e, POLYAREA * dest, PLINE ** src)
 
             /* Remove hole from the contour */
             remove_contour (pa_info->pa, prev, info.result, TRUE);
-
-            /* Add hole as the next on the list to be processed in this very function */
-            info.result->next = *src;
-            *src = info.result;
+            poly_DelContour (&info.result);
           }
           /* End check for kicked out holes */
